src/components/drawer/ComponentDetailPanel.vue

```diff
@@ -58,9 +62,22 @@ function closeDrawer() {
 
 onMounted(() => {
   props.selectionEvent.subscribe((newVal) => {
+    origValue = JSON.parse(JSON.stringify(newVal));
     selectedComponent.value = newVal;
   });
 });
+
+function upsertAttribute(newAttr) {
+  if (!selectedComponent.value.attributes) selectedComponent.value.attributes = [];
+  const attrIndex = selectedComponent.value.attributes.findIndex((a) => a.name === newAttr.name);
+  if (attrIndex >= 0) selectedComponent.value.attributes[attrIndex] = newAttr;
```

Comment:
just update the value.

Reply:
tried it, triggers a re-render loop until the component crashes